Dockerfile

```diff
@@ -1,7 +1,24 @@
 FROM broadinstitute/java-baseimage
 MAINTAINER Broad Institute DSDE <dsde-engineering@broadinstitute.org>
 
-COPY dist/picard.jar /usr/picard/picard.jar
+# Install ant, git for building
+RUN apt-get update && \
+    apt-get --no-install-recommends install -y --force-yes \
+        git \
+        ant && \
+    apt-get clean autoclean && \
+    apt-get autoremove -y
+
+# Assumes Dockerfile lives in root of the git repo. Pull source files into container
+COPY build.xml /usr/picard/build.xml
+COPY src /usr/picard/src
+COPY lib /usr/picard/lib
 WORKDIR /usr/picard
 
-ENTRYPOINT ["java", "-jar", "picard.jar"]
+# Clone out htsjdk. First turn off git ssl verification
+RUN git config --global http.sslVerify false && git clone https://github.com/samtools/htsjdk.git
```

Comment:
Yah, clone-htdjdk uses the ssh clone, and root user in a docker container doesn't have keys.